cmd/generate/cmd_util_test.go

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package main
+
+import (
+	"testing"
+
+	"github.com/nginxinc/nginx-go-crossplane/internal/generator"
+	"github.com/stretchr/testify/require"
+)
+
+//nolint:funlen
+func TestOverrideParser(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name     string
+		input    string
+		expected overrideItem
+		wantErr  bool
+	}{
+		{
+			name:  "normalFormat_pass",
+			input: "location:ngxHTTPMainConf|ngxConfTake12,ngxStreamMainConf",
+
+			expected: overrideItem{
+				directive: "location",
+				masks: []generator.Mask{
+					{"ngxHTTPMainConf", "ngxConfTake12"},
+					{"ngxStreamMainConf"},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:  "withSpaces_pass",
+			input: "hash:ngxHTTPUpsConf | ngxConfTake12, ngxStreamUpsConf | ngxConfTake12",
+			expected: overrideItem{
+				directive: "hash",
+				masks: []generator.Mask{
+					{"ngxHTTPUpsConf", "ngxConfTake12"},
+					{"ngxStreamUpsConf", "ngxConfTake12"},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:    "withoutColon_fail",
+			input:   "hashngxHTTPUpsConf | ngxConfTake12,ngxStreamUpsConf | ngxConfTake12",
+			wantErr: true,
+		},
+		{
+			name:    "colonLeftsideEmpty_fail",
+			input:   " :ngxHTTPUpsConf | ngxConfTake12,ngxStreamUpsConf | ngxConfTake12",
+			wantErr: true,
+		},
+		{
+			name:    "colonRightsideEmpty_fail",
+			input:   "hash:  ",
+			wantErr: true,
+		},
+		{
+			name:    "emptyBitmask_fail",
+			input:   "hash: ngxHTTPUpsConf| ",
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			var parser overrideItem
+			err := parser.UnmarshalText([]byte(tc.input))
```

Comment:
nit: naming: Golang likes short variables names (see https://go.dev/wiki/CodeReviewComments#variable-names), in tests you'll often see two variables `want` and `got` to represent what the test wanted to the code to return, and what was actually returned.

Recommend using something besides "parser"

```suggestion
			var got overrideItem
			err := got.UnmarshalText([]byte(tc.input))
```
or maybe `o`
```suggestion
			var o overrideItem
			err := o.UnmarshalText([]byte(tc.input))
```

Reply:
Get it. Applied